test.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# Prerequisites
+# * https://github.com/ko-build/ko/releases/tag/v0.16.0
+
+set -o nounset
+set -o errexit
+set -o pipefail
+set -o xtrace
+
+: ${VEN_API_KEY?}
+: ${VEN_OWNING_TEAM?}
+
+script_dir=$(cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd)
+root_dir=$(cd "${script_dir}/.." && pwd)
+
+cd "${script_dir}"
+
+export VERSION=0.1.49
+export TERM=dumb
+OCI_BASE=ttl.sh/63773370-0bcf-4ac0-bd42-5515616089ff
+export KO_DOCKER_REPO=$OCI_BASE/images/venafi-agent
+
+ko build . --bare --tags "v${VERSION}"
+helm package deploy/charts/venafi-kubernetes-agent --version "${VERSION}"
+helm push venafi-kubernetes-agent-${VERSION}.tgz "oci://${OCI_BASE}/charts"
+
+kind create cluster || true
+
+kubectl create ns venafi || true
+
+# Pull secret for Venafi OCI registry
+if ! kubectl get secret venafi-image-pull-secret -n venafi; then
+    venctl iam service-accounts registry create \
+           --no-prompts \
+           --owning-team "${VEN_OWNING_TEAM}" \
+           --name "venafi-kubernetes-agent-e2e-registry-${RANDOM}" \
+           --scopes enterprise-cert-manager,enterprise-venafi-issuer,enterprise-approver-policy \
+    | jq '{
+            "apiVersion": "v1",
+            "kind": "Secret",
+            "metadata": {
+              "name": "venafi-image-pull-secret"
+            },
+            "type": "kubernetes.io/dockerconfigjson",
+            "stringData": {
+              ".dockerconfigjson": {
+                "auths": {
+                  "\(.oci_registry)": {
+                    "username": .username,
+                    "password": .password
+                  }
+                }
+              } | tostring
+            }
+          }' \
+    | kubectl create -n venafi -f -
+fi
+
+# Service account credentials for venafi-kubernetes-agent
+if ! kubectl get secret agent-credentials -n venafi; then
+    venctl iam service-account agent create \
+           --no-prompts \
+           --owning-team "${VEN_OWNING_TEAM}" \
+           --name "venafi-kubernetes-agent-e2e-agent-${RANDOM}" \
+    | jq '{
+            "apiVersion": "v1",
+            "kind": "Secret",
+            "metadata": {
+              "name": "agent-credentials"
+            },
+            "stringData": {
+              "privatekey.pem": .private_key,
+              "client-id": .client_id
+            }
+          }' \
+    | kubectl create -n venafi -f -
+fi
+
+# export VENAFI_KUBERNETES_AGENT_CLIENT_ID=$(kubectl get secret -n venafi agent-credentials -o jsonpath='{.data.client-id}' | base64 -d)
+export VENAFI_KUBERNETES_AGENT_CLIENT_ID=""
+venctl components kubernetes apply \
+       --venafi-kubernetes-agent \
+       --venafi-kubernetes-agent-version "$VERSION" \
+       --venafi-kubernetes-agent-values-files "${script_dir}/values.venafi-kubernetes-agent.yaml" \
+       --venafi-kubernetes-agent-custom-image-registry "${OCI_BASE}/images" \
+       --venafi-kubernetes-agent-custom-chart-repository "oci://${OCI_BASE}/charts"
```

Comment:
The next step is to create a VenafiConnection. You can create it using the following command:

```bash
privatekey=$(kubectl get secret -n venafi agent-credentials --template="{{index .data \"privatekey.pem\" | base64decode}}")
clientid=$(kubectl get secret -n venafi agent-credentials --template="{{index .data \"client-id\" | base64decode}}")
accesstoken=$(step crypto jwt sign \
  --key <(sed 's/ PRIVATE KEY/ EC PRIVATE KEY/g' <<<"$privatekey") \
  --aud api.venafi.cloud/v1/oauth/token/serviceaccount \
  --exp $(date -v+30d +"%s") \
  --sub "$clientid" \
  --iss "$clientid" \
    | tee >(step crypto jwt inspect --insecure >/dev/stderr))
```

Then, create the venafi connection:

```bash
kubectl apply -f - <<EOF
apiVersion: jetstack.io/v1alpha1
kind: VenafiConnection
metadata:
  name: test-accesstoken
  namespace: venafi
spec:
  vcp:
    url: https://api.venafi.cloud
    apiKey:
      - secret:
          name: accesstoken
          fields: ["accesstoken"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: venafi-components-reads-accesstoken
  namespace: venafi
rules:
- apiGroups: [ "" ]
  resources: [ "secrets" ]
  verbs: [ "get" ]
  resourceNames: [ "accesstoken" ]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: venafi-components-reads-accesstoken
  namespace: venafi
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: venafi-components-reads-accesstoken
subjects:
- kind: ServiceAccount
  name: venafi-connection
  namespace: venafi
EOF
```

Note that this isn't using the "secretless auth", but this should let you test things. To use the secretless auth, you can't use `venctl` to create the service accounts yet. I'll give you a curl command to do that if you like.

Reply:
I updated the test script to create a signed JWT and exchange it for a bearer token using the method you documented on Venafi internal wiki.